vm/src/stdlib/socket.rs

```diff
@@ -352,6 +362,85 @@ impl SocketRef {
             Err(s) => Err(vm.new_os_error(s.to_string())),
         }
     }
+
+    fn gettimeout(self, vm: &VirtualMachine) -> PyResult {
+        match self.timeout.borrow().as_ref() {
+            Some(duration) => Ok(vm.ctx.new_float(duration.as_secs() as f64)),
+            None => Ok(vm.get_none()),
+        }
+    }
+
+    fn setblocking(self, block: PyObjectRef, vm: &VirtualMachine) -> PyResult<()> {
+        match boolval(vm, block.clone()) {
+            Ok(value) => {
+                if value {
+                    self.timeout.replace(None);
+                } else {
+                    self.timeout.borrow_mut().replace(Duration::from_secs(0));
+                }
+                if self.internal_setblocking(value) {
+                    Ok(())
+                } else {
+                    Err(vm.new_os_error("Set socket blocking error".to_string()))
+                }
+            }
+            Err(_) => Err(vm.new_type_error(format!(
+                "an integer os required (got type {})",
+                &block.class().name.clone()
+            ))),
+        }
+    }
+
+    fn internal_setblocking(&self, block: bool) -> bool {
+        match self.con.borrow_mut().as_mut() {
+            Some(conn) => match conn {
```

Comment:
I would prefer if you implement a `set_nonblocking` on the `Connection` instead of here. Same as all other methods.

Reply:
Done :) It looks great